yt_dlp/extractor/bilibili.py

```diff
@@ -152,11 +178,50 @@ def _get_episodes_from_season(self, ss_id, url):
             note='Downloading season info', query={'season_id': ss_id},
             headers={'Referer': url, **self.geo_verification_headers()})
 
-        for entry in traverse_obj(season_info, (
-                'result', 'main_section', 'episodes',
-                lambda _, v: url_or_none(v['share_url']) and v['id'])):
+        for entry in traverse_obj(season_info, ('result', 'main_section', 'episodes',
+                                                lambda _, v: url_or_none(v['share_url']) and v['id'])):
             yield self.url_result(entry['share_url'], BiliBiliBangumiIE, f'ep{entry["id"]}')
 
+    def _enc_wbi(self, params: dict, video_id=None):
+        if video_id is None:
+            video_id = 0
+        session_data = self._download_json('https://api.bilibili.com/x/web-interface/nav',
+                                           video_id, note='wbi signature...', fatal=False)
+
+        key_from_url = lambda x: x[x.rfind('/') + 1:].split('.')[0]
+        img_key = traverse_obj(
+            session_data, ('data', 'wbi_img', 'img_url', {key_from_url}))
+        sub_key = traverse_obj(
+            session_data, ('data', 'wbi_img', 'sub_url', {key_from_url}))
+
+        mixin_key_enc_tab = [
+            46, 47, 18, 2, 53, 8, 23, 32, 15, 50, 10, 31, 58, 3, 45, 35, 27, 43, 5, 49,
+            33, 9, 42, 19, 29, 28, 14, 39, 12, 38, 41, 13, 37, 48, 7, 16, 24, 55, 40,
+            61, 26, 17, 0, 1, 60, 51, 30, 4, 22, 25, 54, 21, 56, 59, 6, 63, 57, 62, 11,
+            36, 20, 34, 44, 52
+        ]
+        mixin_key = functools.reduce(lambda s, i: s + (img_key + sub_key)[i], mixin_key_enc_tab, '')[:32]
+        params['wts'] = round(time.time())
+        params = dict(sorted(params.items()))
+        params = {
+            k: ''.join(filter(lambda char: char not in "!'()*", str(v)))
+            for k, v
+            in params.items()
+        }
```

Comment:
We dont have to make it a dict only to call `items()` again. Also we can use `'!\'()*'.__contains__` (`'` is required by coding conventions even if we have to escape)
```suggestion
        params = {
            k: ''.join(filter('!\'()*'.__contains__, str(v)))
            for k, v in sorted(params.items())
        }
```

Reply:
You may have misunderstood the meaning of here. The purpose is to remove these symbols instead of just keeping them.

Reply:
Ahh I mixed them up in my head. You are correct we would need `filterfalse`, but I guess at that point its easier to use `lambda char: char not in '!\'()*'` instead.